pkg/datapath/loader/compile.go

```diff
@@ -396,11 +395,13 @@ func Compile(ctx context.Context, src string, out string) error {
 
 // compileTemplate compiles a BPF program generating a template object file.
 func compileTemplate(ctx context.Context, out string, isHost bool) error {
+	debug := option.Config.Opts.IsEnabled(option.DebugCompile)
```

Comment:
same here, this needs to be `debug := option.Config.BPFCompilationDebug || option.Config.Opts.IsEnabled(option.DebugCompile)`